Doc/library/tkinter.rst

```diff
@@ -67,16 +67,49 @@ Or, more often::
    from tkinter import *
 
 
-.. class:: Tk(screenName=None, baseName=None, className='Tk', useTk=1)
-
-   The :class:`Tk` class is instantiated without arguments. This creates a toplevel
-   widget of Tk which usually is the main window of an application. Each instance
-   has its own associated Tcl interpreter.
-
-   .. FIXME: The following keyword arguments are currently recognized:
-
-
-.. function:: Tcl(screenName=None, baseName=None, className='Tk', useTk=0)
+.. class:: Tk(screenName=None, baseName=None, className='Tk', useTk=True, sync=False, use=None)
+
+   Construct a toplevel Tk widget, which is usually the main window of an
+   application.  Each instance has its own associated Tcl interpreter.
+
+   The :class:`Tk` class is typically instantiated using all default values.
+   However, the following keyword arguments are currently recognized:
+
+   *screenName*
+      When given (as a string), sets the :envvar:`DISPLAY` environmental
+      variable. (X11 only)
+   *baseName*
+      Name of the profile file used in :meth:`readprofile`.  By default,
+      *baseName* is derived from the program name (``sys.argv[0]``).
+   *className*
+      Name of the widget class.  Used as a profile file in :meth:`readprofile`
+      and used as the name with which Tcl is invoked (*argv0* in *interp*).
+   *useTk*
+      If ``True``, initialize the Tk subsystem.  The :func:`tkinter.Tcl() <Tcl>`
+      function sets this to ``False``.
+   *sync*
+      If ``True``, execute all X server commands synchronously, so that errors
+      are reported immediately.
+   *use*
+      Specifies the *id* of the main window in which to embed the application,
+      instead of it being created as an independent toplevel window. *id* must
+      be specified in the same way as the value for the -use option for
+      toplevel widgets (that is, it has a form like that returned by
+      :meth:`winfo_id`).
+
+      Note that on some platforms this will only work correctly if *id* refers
+      to a Tk frame or toplevel that has its -container option enabled.
+
+   .. method:: readprofile(self, baseName, className)
```

Comment:
It's documented as an internal function, I'm not sure that we want to document it.

Reply:
@serhiy-storchaka  had mentioned "It may be worth to document reading profiles separately, and only left a short reference here. There is a common between baseName and className, and there are details that should be documented.", so that's why I documented the method.  I wasn't quite sure how else to add what he asked for.

Reply:
Sorry, I didn't mean that `readprofile()` should be documented. It is a private method. But after describing parameters it is worth to document that Tk() reads and interprets files, what are file names, and where they are searched. This creates a security issue, because arbitrary code can be executed in the context and with permissions of the program.

Reply:
Yes, the security issue discussion is at https://bugs.python.org/issue16248.